components/micro-gateway-core/src/main/ballerina/src/gateway/analytics/responsetime_util.bal

```diff
@@ -81,26 +81,29 @@ public function generateRequestResponseExecutionDataEvent(http:Response response
     runtime:InvocationContext invocationContext = runtime:getInvocationContext();
     if (isSecured && invocationContext.attributes.hasKey(AUTHENTICATION_CONTEXT)) {
         AuthenticationContext authContext = <AuthenticationContext>invocationContext.attributes[AUTHENTICATION_CONTEXT];
-        requestResponseExecutionDTO.apiCreator = authContext.apiPublisher;
```

Comment:
Don't we still have to set the api Creator

Reply:
publisher is already known to MGW. Therefore this value always set using the `@gateway:publisher` value in https://github.com/wso2/product-microgateway/pull/956/files#diff-65f9f6b86ad06860d5207a18519e41f7R105